### PR TITLE
sourcekitten: improve test & bump to 0.32.0

### PR DESCRIPTION
**Pull request by jpsim**
As of https://github.com/jpsim/SourceKitten/pull/728 there's now a way to run SourceKitten while avoiding XPC communication with SourceKit.

This should prevent the sandbox violations we were seeing before.

### CODE REVIEW

**Comment by jpsim** [comment]
So this is interesting. This stronger test requires Xcode 13, even though SourceKitten still supports building with Xcode 12. I'll either check to see what version of Xcode is configured before running this additional test, or more likely in the near future SourceKitten will require Xcode 13 to build so I can just unconditionally run this stronger test.

**Comment by SMillerDev** [comment]
AFAIR xcode 13 is installed on all CI nodes so that should be fine

**Comment by jpsim** [comment]
Do you know the installation path? I can set it as `DEVELOPER_DIR` in the test.

File: Formula/sourcekitten.rb [hunk 1]
```diff
@@ -24,8 +24,10 @@ def install
   end
 
   test do
-    # Rewrite test after sandbox issues investigated.
-    # https://github.com/Homebrew/homebrew/pull/50211
     system "#{bin}/sourcekitten", "version"
+    if `xcodebuild -version`.include? "Xcode 13"
+      ENV["IN_PROCESS_SOURCEKIT"] = "YES"
+      system "#{bin}/sourcekitten", "syntax", "--text", "import Foundation // Hello World"
+    end
```
Comment (carlocab) [inline Formula/sourcekitten.rb:31]:
```suggestion
    return if MacOS::Xcode.version < 13

    ENV["IN_PROCESS_SOURCEKIT"] = "YES"
    system "#{bin}/sourcekitten", "syntax", "--text", "import Foundation // Hello World"
```

**Comment by carlocab** [comment]
> Do you know the installation path? I can set it as `DEVELOPER_DIR` in the test. 

You can use `MacOS::Xcode.sdk_path` I think.

**Comment by jpsim** [comment]
> AFAIR xcode 13 is installed on all CI nodes so that should be fine

Xcode 13 requires macOS 11.3+, so it's not installed on the macOS 10.15 nodes.

**Review by carlocab — APPROVED** [review]
Thanks!

**Pull request closed without merging**